Per 208, 206, 216

evanhackett opened the pull request:
Lots of changes here. Sorry for not breaking this into multiple branches with multiple PRs.

I did break things into multiple commits though.

The main things are:

added ability to rename captions.
added text input to caption form, so now a text string can be entered and saved for a caption.
added delete buttons for caption
minor styling tweaks
minor refactor of the url/route structure (this was necessary to make the caption editor page work)

evanhackett committed:
Add caption name and text form

Also changed route urls to better reflect the structure of the paths.
Also added a currentCaption property to the model which will be useful.

evanhackett committed:
add delete button on caption items

Also renamed Scene.current to be Scene.currentScene now that there is
also Scene.currentCaption. Added some minor CSS as well.

evanhackett committed:
add test for caption name and text form

evanhackett committed:
add test for deleteCaption

evanhackett committed:
fix bug regarding caption ids after deletion

JosephJindrich reviewed (changes requested):
The delete caption page is still bugged. After messing around with it, I was able to make captions completely stop appearing when I went to the url page. I would test the delete function a bit more. If I have time I will try and figure it out.

JosephJindrich commented on src/client/src/views/CaptionForm.js:
```diff
@@ -1,11 +1,43 @@
   view: function(vnode) {
     return m('', [
-      m('h1', `Caption ${vnode.attrs.id}`),
+      m('h1', Scene.currentCaption.name ? Scene.currentCaption.name : `Caption ${vnode.attrs.captionId}`),
```
Change this to `m('h1', Scene.currentCaption.name ? Scene.currentCaption.name : 'Caption String Name'),` please

JosephJindrich commented:
@evanhackett If you could, I'd like you to also change caption to caption string. That was the name we decided to give what we've been labeling captions i the code. I'd like to keep the same naming conventions through the project.

JosephJindrich commented on src/client/src/index.js:
```diff
@@ -19,12 +19,12 @@ const CaptionForm = require('./views/CaptionForm')
   '/start': Start,
   '/scenes': Scenes,
-  '/edit-scene/:id': {
+  '/scenes/scene-:id': {
```
Could you change :id to :sceneId like it is in the caption section

evanhackett replied:
Ok I fixed this.

JosephJindrich commented on src/client/src/views/SceneForm.js:
```diff
@@ -9,13 +9,13 @@ module.exports = {
         href: '/scenes',
       }, 'Back To Scene List'),
-      m('h1', Scene.current.name ? Scene.current.name : `Scene ${Scene.current.id}`),
+      m('h1', Scene.currentScene.name ? Scene.currentScene.name : `Scene ${Scene.currentScene.id}`),
```
Make this `m('h1', 'Scene Name'),` please

JosephJindrich commented on src/client/src/views/SceneForm.js:
```diff
@@ -25,34 +25,46 @@ module.exports = {
+            Scene.currentScene.name = e.target.value
           },
-          value: Scene.current.name ? Scene.current.name : `Scene ${Scene.current.id}`
+          value: Scene.currentScene.name ? Scene.currentScene.name : `Scene ${Scene.currentScene.id}`
```
Make this `value: Scene.currentScene.name ? Scene.currentScene.name : ''` please

JosephJindrich commented on src/client/src/views/CaptionForm.js:
```diff
@@ -1,11 +1,43 @@
+          oninput: function (e) {
+            Scene.currentCaption.name = e.target.value
+          },
+            value: Scene.currentCaption.name ? Scene.currentCaption.name : `Caption ${Scene.currentCaption.id}`
```
Change this to `value: Scene.currentCaption.name ? Scene.currentCaption.name : ''` please

evanhackett committed:
renamed id to sceneId

evanhackett committed:
reference changed to global var instead of vnode

For whatever reason, accessing the value through the vnode attribute was
causing problems. Since the same value was in a global var anyway, it is
now accessed with that instead.

JosephJindrich reviewed (approved):
Looks good
